Changes how halloween lighting works slightly

LemonInTheDark opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Fully dropping lighting is a nice effect, but it makes dark spaces pitch. So instead, let's bump it, then drop that slightly more. That dumps the brights and only slightly darkens. I've also lowered saturation which helps make the effect more scalar rather then direct offfsets.

My intent is not to make this totally normal/remove it, since I do want it to impact vibes. Tolerable for 5 days but vibey > workable for a year if that makes sense. 

Closes #79325 (mostly, won't work for the quirk but I am willing to pay that cost)
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

![image](https://github.com/tgstation/tgstation/assets/58055496/0bc71347-772c-4558-a4c3-da412a9ce846)

![image](https://github.com/tgstation/tgstation/assets/58055496/4eb1345f-f6b1-4b28-a9b3-57319c0c41d1)

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: Halloween lighting should no longer black out lavaland. 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

LemonInTheDark committed:
Changes how halloween lighting works slightly

Fully dropping lighting is a nice effect, but it makes dark spaces
pitch. So instead, let's bump it, then drop that slightly more. That
dumps the brights and only slightly darkens. I've also lowered
saturation which helps make the effect more scalar rather then direct
offfsets

Jacquerel commented:
Comparison for people with darkvision, as that was a specific complaint people had WRT shadowpeople.

**Shadowpeople**

No filter:
![image](https://github.com/tgstation/tgstation/assets/7483112/328caf4f-c685-4500-9be8-708fff7b0652)

With this PR:
![image](https://github.com/tgstation/tgstation/assets/7483112/d087a060-89f1-440a-a007-8b35e3041e5c)


**Night Vision Mesons**

No filter:
![image](https://github.com/tgstation/tgstation/assets/7483112/8c8f0a66-ca35-476c-9ce0-2b6a17511273)

With this PR:
![image](https://github.com/tgstation/tgstation/assets/7483112/f2fc73e7-8edf-455b-bf98-e5c3e4525433)

With the proviso that "works on my machine" is always somewhat contentious when it comes to lighting due to varying monitor settings... the Nightmare example is still basically decipherable to me (if harder to distinguish) but Lavaland with Night Vision is still essentially a void. 

Is there anything that can be done to, idk, just reduce or remove the effect while wearing night vision? I imagine that will be a lot more complicated than just blanket-applying the filter to everyone.
Ultimately "just carry a lantern in lavaland" might be fine for a week as long as we know it's an effect, but it _is_ making what is usually a useful mid-late round tool for the job essentially useless.

LemonInTheDark commented:
> Is there anything that can be done to, idk, just reduce or remove the effect while wearing night vision? I imagine that will be a lot more complicated than just blanket-applying the filter to everyone. Ultimately "just carry a lantern in lavaland" might be fine for a week as long as we know it's an effect, but it _is_ making what is usually a useful mid-late round tool for the job essentially useless.

Yeah totally. I can drop the drop effect with nightvision on, which'll solve our troubles here. Reasonably easy to do.
Also mesons are a roundstart item for mining, you're meant to use a light with them regardless since they don't show mobs so well (this was intentional). Not sure what you mean by mid/late round tool.

LemonInTheDark committed:
Adds cutoff support, refactors things a bit

Fixes a bug in filter's modify proc that dropped priority, removed the
sync lighting color proc and added a signal sent to hud when sight is
updated, hooks to that instead.

LemonInTheDark commented:
Nightvision now cuts down the drop effect to its minimum possible to not cut into lighting (roughly not exact), and we instead do the dimming with multiplication.
It looks worse, but it's close enough that it's fine.

Jacquerel commented:
Afraid that with your new branch I _still_ can't see shit with night vision on lavaland, but also on top of that it runtimes when you observe

san7890 commented:
halloween is over do you wanna keep tweaking in anticipation of next year or shove it for later

LemonInTheDark commented:
Honestly I might just remove it, it causes typical issues like shittly screen configuration, different environments, and just less then perfect eyesight. I have ideas for the effect but we should do somethin different for halloween

san7890 commented:
yeah i played a shift with it on yesterday and i had to keep my flashlights on around the clock just so i could see stuff right in front of my character, which is kinda spooky but is also just kinda weird when the captain's office is so dimly lit they need a flashlight to see much of anything
